spec/models/program_session_spec.rb

```diff
@@ -1,5 +1,99 @@
-  pending "add some examples to (or delete) #{__FILE__}"
+describe ProgramSession do
+
+  let(:proposal) { create(:proposal, :with_two_speakers) }
+
+  it "responds to .create_from_proposal" do
+    expect(ProgramSession).to respond_to(:create_from_proposal)
+  end
+
+  describe ".create_from_proposal" do
+    context "creates a program session from a proposal" do
+
+      it "program session and proposal share the same title" do
+        session = ProgramSession.create_from_proposal(proposal)
+
+        expect(session.title).to eq(proposal.title)
+      end
+
+      it "program session and proposal share the same abstract" do
+        session = ProgramSession.create_from_proposal(proposal)
+
+        expect(session.abstract).to eq(proposal.abstract)
+      end
+
+      it "program session and proposal share the same event" do
+        session = ProgramSession.create_from_proposal(proposal)
+
+        expect(session.event_id).to eq(proposal.event_id)
+      end
+
+      it "program session and proposal share the same session format" do
+        session = ProgramSession.create_from_proposal(proposal)
+
+        expect(session.session_format_id).to eq(proposal.session_format_id)
+      end
+
+      it "program session and proposal share the same track" do
+        session = ProgramSession.create_from_proposal(proposal)
+
+        expect(session.track_id).to eq(proposal.track_id)
+      end
+
+      it "program session has proposal id" do
+        session = ProgramSession.create_from_proposal(proposal)
+
+        expect(session.proposal_id).to eq(proposal.id)
+      end
+
+      it "program session is active" do
+        session = ProgramSession.create_from_proposal(proposal)
+
+        expect(session.state).to eq("active")
+      end
+
+      it "sets program session id for all speakers" do
+        session = ProgramSession.create_from_proposal(proposal)
+
+        proposal.speakers.each do |speaker|
+          expect(speaker.program_session_id).to eq(session.id)
+        end
+      end
+
+      it "sets speaker_name on each speaker" do
+        proposal.speakers.each do |speaker|
+          expect(speaker.speaker_name).to eq(nil)
+        end
+
+        ProgramSession.create_from_proposal(proposal)
+
+        proposal.speakers.each do |speaker|
+          expect(speaker.speaker_name).to eq(speaker.user.name)
+        end
+      end
+
+      it "sets speaker_email on each speaker" do
+        proposal.speakers.each do |speaker|
+          expect(speaker.speaker_email).to eq(nil)
+        end
+
+        ProgramSession.create_from_proposal(proposal)
+
+        proposal.speakers.each do |speaker|
+          expect(speaker.speaker_email).to eq(speaker.user.email)
+        end
+      end
+
+      it "each speaker still has proposal id" do
+        ProgramSession.create_from_proposal(proposal)
+
+        proposal.speakers.each do |speaker|
+          expect(speaker.proposal_id).to eq(proposal.id)
+        end
+      end
+
```

Comment:
Go ahead and throw in a couple more at the end here:
  speaker_name won't get overwritten if it already has a value
  speaker_email won't get overwritten if it already has a value